worker/codalabworker/local_run/local_run_state.py

```diff
@@ -392,12 +412,18 @@ def _transition_from_CLEANING_UP(self, run_state):
         """
         bundle_uuid = run_state.bundle['uuid']
         if run_state.container_id is not None:
-            while True:
+            while docker_utils.container_exists(run_state.container):
                 try:
                     finished, _, _ = docker_utils.check_finished(run_state.container)
                     if finished:
                         run_state.container.remove(force=True)
+                        run_state = run_state._replace(container=None, container_id=None)
```

Comment:
@bkgoksel why was this line of code needed?

Reply:
It's been a while since I opened this PR, but basically we use None-checks on these fields to decide whether this run still has a container attached to it, so the right time to set these back to None is immediately after we're sure we managed to remove the container without running into issues, which is here.